utils/oomparser/oomparser_test.go

```diff
@@ -0,0 +1,164 @@
+	couldParseLine, err := getProcessNamePid(startLine, currentOomInstance)
+	if err != nil {
+		t.Errorf("bad line fed to getProcessNamePid should yield no error, but had error %v", err)
+	}
+	if couldParseLine {
+		t.Errorf("bad line fed to getProcessNamePid should return false but returned %s", couldParseLine)
+	}
+
+	correctTime, err := time.Parse(time.Stamp, "Jan 21 22:01:49")
+	couldParseLine, err = getProcessNamePid(endLine, currentOomInstance)
+	if err != nil {
+		t.Errorf("good line fed to getProcessNamePid should yield no error, but had error %v", err)
+	}
+	if !couldParseLine {
+		t.Errorf("good line fed to getProcessNamePid should return true but returned %s", couldParseLine)
+	}
+	if currentOomInstance.ProcessName != "evilprogram2" {
+		t.Errorf("getProcessNamePid should have set processName to evilprogram2, not %s", currentOomInstance.ProcessName)
+	}
+	if currentOomInstance.Pid != 19667 {
+		t.Errorf("getProcessNamePid should have set PID to 19667, not %d", currentOomInstance.Pid)
+	}
+	if !correctTime.Equal(currentOomInstance.TimeOfDeath) {
+		t.Errorf("getProcessNamePid should have set date to %v, not %v", correctTime, currentOomInstance.Pid)
+	}
+}
+
+func TestCheckIfStartOfMessages(t *testing.T) {
+	couldParseLine, err := checkIfStartOfOomMessages(endLine)
+	if err != nil {
+		t.Errorf("bad line fed to checkIfStartOfMessages should yield no error, but had error %v", err)
+	}
+	if couldParseLine {
+		t.Errorf("bad line fed to checkIfStartOfMessages should return false but returned %s", couldParseLine)
+	}
+
+	couldParseLine, err = checkIfStartOfOomMessages(startLine)
+	if err != nil {
+		t.Errorf("start line fed to checkIfStartOfMessages should yield no error, but had error %v", err)
+	}
+	if !couldParseLine {
+		t.Errorf("start line fed to checkIfStartOfMessages should return true but returned %s", couldParseLine)
+	}
+}
+
+func TestAnalyzeLines(t *testing.T) {
+	outStream := make(chan *OomInstance)
+	oomLog := new(OomParser)
+	oomLog.systemFile = "testmessagefeed.txt"
+	file, err := os.Open(oomLog.systemFile)
+	if err != nil {
+		t.Errorf("cgoouldn't open test log: %v", err)
+	}
+	correctTime, err := time.Parse(time.Stamp, "Jan 21 22:01:49")
+	oomCheckInstance := OomInstance{
+		Pid:         19667,
+		ProcessName: "evilprogram5",
+		TimeOfDeath: correctTime,
+	}
+	timeout := make(chan bool, 1)
+	go func() {
+		time.Sleep(1 * time.Second)
+		timeout <- true
+	}()
+	go oomLog.analyzeLines(file, outStream)
+	select {
+	case oomInstance := <-outStream:
+		if oomCheckInstance != *oomInstance {
+			t.Errorf("wrong instance returned. Expected %v and got %v",
+				oomCheckInstance, oomInstance)
+		}
+	case <-timeout:
+		t.Error(
+			"timeout happened before oomInstance was found in test file")
+	}
+}
+
+func TestStreamOoms(t *testing.T) {
+	outStream := make(chan *OomInstance)
+	oomLog := new(OomParser)
+	oomLog.systemFile = "testmessagefeed.txt"
+
+	correctTime, err := time.Parse(time.Stamp, "Jan 21 22:01:49")
```

Comment:
nit: Maybe have a helper that creates the expected OomInstance. We can use it in both of these tests and some of the ones above.